module/modmanager/manager.go

```diff
@@ -144,8 +151,19 @@ func (mgr *Manager) add(ctx context.Context, conf config.Module, conn *grpc.Clie
 		return nil
 	}
 
+	var moduleDataDir string
+	if mgr.moduleDataParentDirectory != "" {
+		moduleDataDir = filepath.Join(mgr.moduleDataParentDirectory, conf.Name)
+		// safety check to prevent exiting the moduleDataDirectory
+		if !strings.HasPrefix(filepath.Clean(moduleDataDir), filepath.Clean(mgr.moduleDataParentDirectory)) {
+			return errors.Errorf("module %q would have a data directory %q outside of the module data directory %q",
+				conf.Name, moduleDataDir, mgr.moduleDataParentDirectory)
+		}
+	}
+
 	mod := &module{
 		cfg:       conf,
+		dataDir:   moduleDataDir,
```

Comment:
Can we please log the value of `moduleDataDir`? Maybe it's not necessary if it's getting logged as part of starting a module process.

I just want to make sure that if module startup depends on finding something in its data dir, we don't have to guess what the program thinks is the data dir.

Reply:
Adding debug log when creating the dir.

Reply:
Added.